mozci/task.py

```diff
@@ -19,6 +21,22 @@ class Status(Enum):
     INTERMITTENT = 2
 
 
+# We only want to warn about bad groups once.
+bad_group_warned = False
+
+
+def is_bad_group(task_id, group):
+    global bad_group_warned
+
+    bad_group = os.path.isabs(group) or group.startswith("file://") or group.startswith("Z:")
+
+    if not bad_group_warned and (bad_group or "\\" in group):
+        bad_group_warned = True
+        logger.warning(f"Bad group name in task {task_id}: {group}")
```

Comment:
Rather than logging here, why not collect all "bad groups" and log the total number of them after the fact?
```
bad_tasks = {<task id>}
bad_groups = {<groups>}
...
self.logger.warning(f"{len(bad_tasks)} tasks contain bad groups:\n{\n.join(list(bad_groups)[:4]}\nand {len(bad_groups)-4} more")
```

Reply:
The problem is, when do we log them?

Reply:
At the end of `__post_init__` ?

Reply:
I guess it could get called multiple times, so you'd have to clear the globals after each log..

Reply:
Oh, nvm, I see what you're saying.. I guess it would have to be in the Push class. Yeah, maybe not worth it.